test/Data/String/ConversionsSpec.hs

```diff
@@ -10,11 +11,184 @@ import           Data.String.Conversions
+        property $ \(s :: String) ->
+          s `shouldBe` cs (toString s)
+
+      it "converts String -> StrictText -> String" $ do
+        property $ \(s :: String) ->
+          s `shouldBe` cs (toStrictText s)
+
+      it "converts String -> LazyText -> String" $ do
+        property $ \(s :: String) ->
+          s `shouldBe` cs (toLazyText s)
+
+      it "converts String -> StrictByteString -> String" $ do
+        property $ \(s :: String) ->
+          s `shouldBe` cs (toStrictByteString s)
+
+      it "converts String -> LazyByteString -> String" $ do
+        property $ \(s :: String) ->
+          s `shouldBe` cs (toLazyByteString s)
+
+      it "converts StrictText -> String -> StrictText" $ do
+        property $ \(s :: StrictText) ->
+          s `shouldBe` cs (toString s)
+
+      it "converts StrictText -> StrictText -> StrictText" $ do
+        property $ \(s :: StrictText) ->
+          s `shouldBe` cs (toStrictText s)
+
+      it "converts StrictText -> LazyText -> StrictText" $ do
+        property $ \(s :: StrictText) ->
+          s `shouldBe` cs (toLazyText s)
+
+      it "converts StrictText -> StrictByteString -> StrictText" $ do
+        property $ \(s :: StrictText) ->
+          s `shouldBe` cs (toStrictByteString s)
+
+      it "converts StrictText -> LazyByteString -> StrictText" $ do
+        property $ \(s :: StrictText) ->
+          s `shouldBe` cs (toLazyByteString s)
+
+      it "converts LazyText -> String -> LazyText" $ do
+        property $ \(s :: LazyText) ->
+          s `shouldBe` cs (toString s)
+
+      it "converts LazyText -> StrictText -> LazyText" $ do
+        property $ \(s :: LazyText) ->
+          s `shouldBe` cs (toStrictText s)
+
+      it "converts LazyText -> LazyText -> LazyText" $ do
+        property $ \(s :: LazyText) ->
+          s `shouldBe` cs (toLazyText s)
+
+      it "converts LazyText -> StrictByteString -> LazyText" $ do
+        property $ \(s :: LazyText) ->
+          s `shouldBe` cs (toStrictByteString s)
+
+      it "converts LazyText -> LazyByteString -> LazyText" $ do
+        property $ \(s :: LazyText) ->
+          s `shouldBe` cs (toLazyByteString s)
+
+      it "converts StrictByteString -> StrictByteString -> StrictByteString" $ do
+        property $ \(s :: StrictByteString) ->
+          s `shouldBe` cs (toStrictByteString s)
+
+      it "converts StrictByteString -> LazyByteString -> StrictByteString" $ do
+        property $ \(s :: StrictByteString) ->
+          s `shouldBe` cs (toLazyByteString s)
+
+      it "converts LazyByteString -> StrictByteString -> LazyByteString" $ do
+        property $ \(s :: LazyByteString) ->
+          s `shouldBe` cs (toStrictByteString s)
+
+      it "converts LazyByteString -> LazyByteString -> LazyByteString" $ do
+        property $ \(s :: LazyByteString) ->
+          s `shouldBe` cs (toLazyByteString s)
+
+    context "without crashing" $ do
+      it "converts String -> String" $ do
```

Comment:
Similarly to https://github.com/soenkehahn/string-conversions/pull/26/files#r842115678 could we have subsections for the groups of tests that test conversions from the same type?